# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/MotionPathUtils.h
# Commit: 29fa69db2655
# Full Hash: 29fa69db2655e35810d121a98dabb0a6e3af6d8d
# Author: Boris Chiou <boris.chiou@gmail.com>
# Date: 2023-05-26 04:06:55
# Regressor Bug: 1821450
# File Overlap Count: 1
# Description:
#   Bug 1821450 - Update the calculation of contain keyword in ray(). r=emilio
#   
#   Per the spec change, when using contain keyword, the path’s length is reduced
#   by half the width or half the height of the element’s border box, whichever is
#   larger, and floored at zero.
# ==============================================================================

diff -r e6d63c2a651d -r 29fa69db2655 layout/base/MotionPathUtils.h
--- a/layout/base/MotionPathUtils.h	Fri May 26 00:13:35 2023 +0000
+++ b/layout/base/MotionPathUtils.h	Fri May 26 00:14:36 2023 +0000
@@ -41,13 +41,17 @@
   CSSPoint mInitialPosition;
   // The rect of the containing block.
   CSSRect mContainingBlockRect;
+  // The size of its border-box in CSS Layout. If it's in SVG layout, this is
+  // the size of view box.
+  CSSSize mBorderBoxSize;
 
   RayReferenceData() = default;
   explicit RayReferenceData(const nsIFrame* aFrame);
 
   bool operator==(const RayReferenceData& aOther) const {
     return mInitialPosition == aOther.mInitialPosition &&
-           mContainingBlockRect == aOther.mContainingBlockRect;
+           mContainingBlockRect == aOther.mContainingBlockRect &&
+           mBorderBoxSize == aOther.mBorderBoxSize;
   }
 };
 